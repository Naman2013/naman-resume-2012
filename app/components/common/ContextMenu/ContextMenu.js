import React, { Component } from 'react';
import PropTypes from 'prop-types';
import VanillaButton from 'atoms/VanillaButton';
import Dots from 'components/common/icons/Dots';

class ContextMenu extends Component {
  state = {
    isOpen: false,
  };

  toggle = () => { this.setState(prevState => ({ isOpen: !prevState.isOpen })); }

  render() {
    return (
      <div className="root">
        <VanillaButton
          handleClick={this.toggle}
          theme={{ width: '100px', height: '100px' }}
        >
          <Dots />
        </VanillaButton>
      </div>
    );
  }
}

export default ContextMenu;
