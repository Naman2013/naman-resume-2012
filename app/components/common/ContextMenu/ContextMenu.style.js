import css from 'styled-jsx/css';
import { primaryFont } from 'styles/variables/fonts';
import { faintGray, astronaut } from 'styles/variables/colors_tiles_v4';

export const CONTAINER_WIDTH = 440;

export default css`
  .root {
    width: 100px;
    overflow: hidden;
  }

  .menu-container {
    position: absolute;
    padding: 0;
    margin: 0;
    top: 110px;
    width: ${Math.abs(CONTAINER_WIDTH)}px;
    height: 100%;
    background-color: ${faintGray};
    transition: right ease-out 0.15s;
  }

  .header-container {
    margin: 0;
    padding: 40px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    color: ${astronaut};
    background-color: white;
  }

  .context-header {
    margin: 0;
    padding: 0;
    font-family: ${primaryFont};
    font-size: 12px;
    text-transform: uppercase;
    letter-spacing: 1.5px;
    font-weight: 800;
  }

  .available-sections {
    margin: 0;
    padding: 0;
    font-family: ${primaryFont};
    font-size: 12px;
  }
`;
