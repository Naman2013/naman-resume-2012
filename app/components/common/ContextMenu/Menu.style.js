import css from 'styled-jsx/css';

export default css`
.menu-list {
  margin: 0;
  padding: 0;
  padding-top: 10px;
}
`;
