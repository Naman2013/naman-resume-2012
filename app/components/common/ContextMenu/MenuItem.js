import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import MiraDot from 'atoms/icons/MiraDot';
import style from './MenuItem.style';

const MenuItem = ({ title, linkURL, iconUrl }) => (
  <Fragment>
    <li className="menu-item">
      <span className="item-container">
        <Link activeClassName="active" to={linkURL}>
          <span className="dot-container">
            <MiraDot />
          </span>
          {title}
        </Link>
        {iconUrl && (
          <div className="iconContainer">
            <img className="icon" src={iconUrl} />
          </div>
        )}
      </span>
    </li>
    <style jsx>{style}</style>
  </Fragment>
);

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  linkURL: PropTypes.string.isRequired,
  iconUrl: PropTypes.string,
};

MenuItem.defaultProps = {
  iconUrl: null,
};

export default MenuItem;
