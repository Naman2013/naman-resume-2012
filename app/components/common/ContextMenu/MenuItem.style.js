import css from 'styled-jsx/css';
import {
  astronaut,
  geyser,
  nightfall,
} from 'app/styles/variables/colors_tiles_v4';
import { secondaryFont } from 'app/styles/variables/fonts';

export default css`
  .menu-item {
    position: relative;
    margin: 0;
    padding: 0;
    margin-left: 70px;
    width: 75%;
    color: ${astronaut};
  }

  .dot-container {
    position: absolute;
    left: -50px;
    top: 43px;
  }

  .item-container {
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid ${geyser};
  }

  .iconContainer {
    width: 50px;
    height: 50px;
    background-color: ${nightfall};
    border-radius: 50%;
    margin-top: 20px;
  }

  .icon {
    width: 34px;
    height: 34px;
    position: relative;
    top: 8px;
    left: 8px;
  }

  .menu-item :global(a) {
    display: block;
    padding-bottom: 30px;
    padding-top: 40px;
    margin-left: -15px;
    padding-left: 20px;
    font-family: ${secondaryFont};
    text-transform: capitalize;
    text-decoration: none;
    color: ${astronaut};
  }

  .menu-item :global(a .dot-container) {
    opacity: 0;
  }

  .menu-item :global(a.active .dot-container) {
    opacity: 1;
  }
`;
