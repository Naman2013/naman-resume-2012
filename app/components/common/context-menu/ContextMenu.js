import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { white } from '../../../styles/variables/colors';

const {
  bool,
  number,
  string,
  func,
} = PropTypes;

class ContextMenu extends Component {

  static propTypes = {
    menuHeight: number,
    menuWidth: number,
    distanceFromTrigger: number,
    backgroundColor: string,
    onShow: func,
  }

  static defaultProps = {
    // how tall the menu will be (in pixels)
    menuHeight: 300,
    // how wide the menu will be (in pixels)
    menuWidth: 200,
    // padding from the trigger button
    distanceFromTrigger: 15,
    // background color for menu
    backgroundColor: white,
    onShow: null,
  };

  state = {
    triggerElement: null,
    left: -(this.props.menuWidth + this.props.distanceFromTrigger),
    top: -(this.props.menuHeight / 2),
    showMenu: false,
  }

  handleContextClick = (e) => {
    const { onShow } = this.props;
    const { showMenu } = this.state;

    if (!showMenu) {
      onShow();
    }

    this.setState({
      showMenu: !showMenu,
    });
  }

  hideMenu = (e) => {
    this.setState({
      showMenu: false,
    });
  }

  render() {
    const {
      menuHeight,
      menuWidth,
      backgroundColor,
      children,
      className,
    } = this.props;

    const { left, top, showMenu } = this.state;

    const menuRootStyle = {
      position: 'absolute',
      top: `${top}px`,
      left: `${left}px`,
      height: `${menuHeight}px`,
      width: `${menuWidth}px`,
      backgroundColor,
    };

    const containerClass = classnames({
      notShown: !showMenu,
    });

    const rootClasses = classnames({
      [className]: !!className,
    });

    return (
      <div
        className={containerClass}
      >
        <div
          className="arrow-right"
        />
        <div
          onMouseLeave={this.hideMenu}
          style={menuRootStyle}
          className={rootClasses}
        >
          {children}
        </div>
        <style jsx>{`

          .notShown {
            visibility: hidden;
          }

          .arrow-right {
            position: absolute;
            left: 15px;
            width: 0;
            height: 0;
            border-top: 20px solid transparent;
            border-bottom: 20px solid transparent;
            border-left: 20px solid ${white};
          }

        `}</style>
      </div>
    );
  }
}

export default ContextMenu;
