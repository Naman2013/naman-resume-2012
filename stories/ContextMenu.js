import React from 'react';
import { storiesOf } from '@storybook/react';
import ContextMenu from '../app/components/common/ContextMenu';

storiesOf('ContextMenu', module)
  .add('Starting closed', () => (
    <div style={{ overFlow: 'hidden', float: 'right' }}>
      <ContextMenu />
    </div>
  ));
