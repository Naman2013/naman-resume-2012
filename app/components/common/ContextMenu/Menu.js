import React from 'react';
import PropTypes from 'prop-types';
import MenuItem from './MenuItem';
import style from './Menu.style';

const Menu = ({ list }) => (
  <ol className="menu-list">
    {list.map(item => (<MenuItem key={`context-menu-${item.title}`} {...item} />))}
    <style jsx>{style}</style>
  </ol>
);

Menu.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    linkURL: PropTypes.string.isRequired,
  })).isRequired,
};

export default Menu;
