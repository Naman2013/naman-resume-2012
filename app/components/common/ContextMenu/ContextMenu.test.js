import React from 'react';
import { shallow } from 'enzyme';
import ContextMenu from './ContextMenu';

describe('ContextMenu', () => {
  const shallowWrapper = shallow(<ContextMenu />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
