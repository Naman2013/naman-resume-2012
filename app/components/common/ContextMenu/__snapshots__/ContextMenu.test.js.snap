// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContextMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <VanillaButton
        handleClick={[Function]}
        theme={
          Object {
            "height": "100px",
            "width": "100px",
          }
        }
      >
        <Dots
          height={4}
          theme={
            Object {
              "circleColor": "white",
            }
          }
          width={18}
        />
      </VanillaButton>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Dots
          height={4}
          theme={
            Object {
              "circleColor": "white",
            }
          }
          width={18}
        />,
        "handleClick": [Function],
        "theme": Object {
          "height": "100px",
          "width": "100px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": 4,
          "theme": Object {
            "circleColor": "white",
          },
          "width": 18,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <VanillaButton
          handleClick={[Function]}
          theme={
            Object {
              "height": "100px",
              "width": "100px",
            }
          }
        >
          <Dots
            height={4}
            theme={
              Object {
                "circleColor": "white",
              }
            }
            width={18}
          />
        </VanillaButton>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Dots
            height={4}
            theme={
              Object {
                "circleColor": "white",
              }
            }
            width={18}
          />,
          "handleClick": [Function],
          "theme": Object {
            "height": "100px",
            "width": "100px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 4,
            "theme": Object {
              "circleColor": "white",
            },
            "width": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
